File: BreakOut3b/BOSceneDelegate.m
//
//  BOSceneDelegate.m
//  BreakOut3b
//

#import "BOSceneDelegate.h"
#import "BOSettingsTableViewController.h"

@implementation BOSceneDelegate

- (void)scene:(UIScene *)scene willConnectToSession:(UISceneSession *)session options:(UISceneConnectionOptions *)connectionOptions
{
    if (![scene isKindOfClass:[UIWindowScene class]]) {
        return;
    }

    UIWindowScene *windowScene = (UIWindowScene *)scene;
    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"MainStoryboard" bundle:nil];
    UIViewController *rootViewController = [storyboard instantiateInitialViewController];

    self.window = [[UIWindow alloc] initWithWindowScene:windowScene];
    self.window.rootViewController = rootViewController;
    [self.window makeKeyAndVisible];

    UINavigationController *navigationController = (UINavigationController *)rootViewController;
    BOSettingsTableViewController *settingsViewController = (BOSettingsTableViewController *)navigationController.topViewController;
    settingsViewController.users = [[NSMutableArray alloc] init];
}

- (void)sceneDidDisconnect:(UIScene *)scene
{
}

- (void)sceneDidBecomeActive:(UIScene *)scene
{
}

- (void)sceneWillResignActive:(UIScene *)scene
{
}

- (void)sceneWillEnterForeground:(UIScene *)scene
{
}

- (void)sceneDidEnterBackground:(UIScene *)scene
{
}

@end

File: BreakOut3b/BOSceneDelegate.swift
import UIKit

@objc(BOSceneDelegate)
final class BOSceneDelegate: UIResponder, UIWindowSceneDelegate {
    var window: UIWindow?

    func scene(
        _ scene: UIScene,
        willConnectTo session: UISceneSession,
        options connectionOptions: UIScene.ConnectionOptions
    ) {
        guard let windowScene = scene as? UIWindowScene else {
            return
        }

        let storyboard = UIStoryboard(name: "MainStoryboard", bundle: nil)
        let rootViewController = storyboard.instantiateInitialViewController()

        let window = UIWindow(windowScene: windowScene)
        window.rootViewController = rootViewController
        window.makeKeyAndVisible()
        self.window = window

        if let navigationController = rootViewController as? UINavigationController,
           let settingsViewController = navigationController.topViewController as? BOSettingsTableViewController {
            settingsViewController.users = []
        }
    }
}
